python/tskit/metadata.py

```diff
@@ -146,15 +334,20 @@ def __init__(self, schema: Optional[dict]) -> None:
             try:
                 TSKITMetadataSchemaValidator.check_schema(schema)
             except jsonschema.exceptions.SchemaError as ve:
-                raise exceptions.MetadataSchemaValidationError from ve
-            codec = schema["codec"]
+                raise exceptions.MetadataSchemaValidationError(str(ve)) from ve
             try:
-                codec_instance = codec_registry[codec](schema)
+                codec_cls = codec_registry[schema["codec"]]
             except KeyError:
                 raise exceptions.MetadataSchemaValidationError(
                     f"Unrecognised metadata codec '{schema['codec']}'. "
                     f"Valid options are {str(list(codec_registry.keys()))}."
                 )
+            codec_instance = codec_cls(schema)
+            # If the codec has modified the schema on __init__, get it back.
```

Comment:
This is strange - need a bit more explanation of what's going on in the comment here.

Reply:
Refactored to make intent clear in https://github.com/tskit-dev/tskit/pull/601/commits/0617b6087274df382489b3b47e181f2e4505e823